test/stdlib/Method_test.rb

```diff
@@ -1,89 +1,181 @@
-require_relative "test_helper"
+require_relative 'test_helper'
 
-class MethodTest < StdlibTest
-  target Method
+class MethodInstanceTest < Test::Unit::TestCase
+  include TestHelper
 
-  class Foo
-    def foo
-    end
+  testing '::Method'
 
-    def foo_with_args(*, **)
-    end
+  def self.takes_anything(*, **, &b)
+    :hello
+  end
 
-    def foo_with_many_args(x, y=42, *other, k_x:, k_y: 42, **k_other, &b)
-    end
+  METHOD = singleton_method(:takes_anything)
 
-    def foo_with_arg_and_rest(x, *)
+  def test_op_eq(method: :==)
+    with_untyped.and METHOD do |untyped|
+      assert_send_type  '(untyped) -> bool',
+                        METHOD, method, untyped
     end
   end
 
-  class Bar < Foo
-    def foo
-    end
+  def test_eql?
+    test_op_eq(method: :eql?)
   end
 
-  def test_to_proc
-    Foo.new.method(:foo).to_proc
+  def test_hash
+    assert_send_type  '() -> Integer',
+                      METHOD, :hash
   end
 
-  def test_call
-    Foo.new.method(:foo).call
-    Foo.new.method(:foo_with_args).call(1)
-  end
+  def test_dup
+    omit_if RUBY_VERSION < '3.4'
```

Comment:
`Method#dup` was only added in ruby 3.4; should we do an `%a{rbs:since:3.4}` to the definition? probably not.